torch/csrc/jit/script/compiler.cpp

```diff
@@ -1345,33 +1354,46 @@ struct to_ir {
 
       emitStatements(body);
 
-      // Also emit the conditional
-      cond_val = (cond) ? emitCond(cond.value())
-                        : graph->insertConstant(true, nullptr, range);
-      body_block->registerOutput(cond_val);
-      auto body_frame = popFrame();
-      auto outer_frame = environment_stack;
-
-      // Add block outputs to correspond to each captured input
-      // some of these will be removed.
-      for (const auto& x : body_frame->captured_inputs) {
-        auto fv = body_frame->getValueInThisFrame(range, x);
-        body_block->registerOutput(fv);
-      }
+      // If a variable was mutated and defined in the enclosing scope,
```

Comment:
Doing this pretty much removes the entire benefit of inserting load/store in the first place. Now people need to have outputs to the block _and_ keep load and store correct.